incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1670,9 +1677,35 @@ else if (length > MAXIMUM_CLIENT_ID_LENGTH)
                 serverDefinedKeepAlive = keepAlive != connect.keepAlive();
                 keepAliveTimeout = Math.round(TimeUnit.SECONDS.toMillis(keepAlive) * 1.5);
                 doSignalKeepAliveTimeout();
+
+                long exchangeAuth = authorization;
+                if (guard != null)
+                {
+                    final String username = payload.username != null ? payload.username.asString() : null;
+                    final String password = payload.password != null ?
+                        payload.password.bytes().get((b, o, m) -> b.getStringWithoutLengthUtf8(o, m - o)) : null;
+                    final String credentialsMatch = credentials.apply(username, password);
```

Comment:
We know it is just one or the other based on the options config, so seems like this should be directly targeted at the correct connect property, not checking to see if they are set or not, agree?